util/timeline_filter/timelineFilter.py

```diff
@@ -149,43 +142,75 @@ def dump(self, filepath, type):
                         else:
                             f.write("<dyn:event time=\"" + str(event.time) + "\" modelName=\"" + event.model+ "\" message=\"" + event.event+ "\" priority=\"" + event.priority + "\"/>\n")
                     except UnicodeEncodeError:
+                        fixEncoding(event)
                         if event.priority == None:
                             f.write("<dyn:event time=\"" + str(event.time).encode('iso8859-1') + "\" modelName=\"" + event.mode.encode('iso8859-1')+ "\" message=\"" + event.event.encode('iso8859-1') + "\"/>\n")
                         else:
                             f.write("<dyn:event time=\"" + str(event.time).encode('iso8859-1') + "\" modelName=\"" + event.model.encode('iso8859-1')+ "\" message=\"" + event.event.encode('iso8859-1')+ "\" priority=\"" + event.priority.encode('iso8859-1') + "\"/>\n")
             f.write("</dyn:timeline>\n")
             f.close()
 
-def read_txt(filepath):
-    timeline = Timeline()
-    f=open(filepath, "r")
+    def filterTimeLine(self, filename, dicOppositeEvents, modelsToKeep, outputName = None):
+        self.filter_useless_events(dicOppositeEvents)
+        self.filter_model(modelsToKeep)
+        if getFileType(filename) == "TXT":
+            self.dump(os.path.join(os.path.dirname(filename), "filtered_timeline.log"), getFileType(filename))
+        elif getFileType(filename) == "XML":
+            if outputName is None:
+                self.dump(os.path.join(os.path.dirname(filename), "filtered_timeline.xml"), getFileType(filename))
+            else:
+                self.dump(os.path.join(os.path.dirname(filename), outputName), getFileType(filename))
+
+def fixEncoding(event):
+    if "<" in event.event:
+        event.event = event.event.replace("<", "&lt;")
+    if ">" in event.event:
+        event.event = event.event.replace(">", "&gt;")
+
+def getFileType(filename):
+    if zipfile.is_zipfile(filename):
+        return "ZIP"
+    elif os.path.basename(filename).endswith(".log") or os.path.basename(filename).endswith(".txt"):
+        return "TXT"
+    elif os.path.basename(filename).endswith(".xml"):
+        return "XML"
+    else:
+        print("[ERROR] unrecognized file extension.")
+        exit(1)
 
-    for line in f.readlines():
-        array = line.split('|')
-        if (len(array) < 3):
-            continue
-        event = Event()
-        time = float(array[0].strip())
-        event.time = time
-        event.model = array[1].strip()
-        event.event = array[2].rstrip().lstrip()
-        if (len(array) >= 4):
-            event.priority = array[3].rstrip().lstrip()
-        timeline.add_event(event)
-    f.close()
-    return timeline
+def checkEnvVar():
+    if os.getenv("DYNAWO_RESOURCES_DIR") is None:
+        print("environment variable DYNAWO_RESOURCES_DIR needs to be defined")
```

Comment:
Maybe we could start the message with [ERROR]?

Reply:
done, thx